daemon/api_base_test.go

```diff
@@ -242,7 +242,7 @@ func (s *apiBaseSuite) SetUpTest(c *check.C) {
 	s.AddCleanup(daemon.MockSystemUserFromRequest(func(r *http.Request) (*user.User, error) {
 		if s.authUser != nil {
 			return &user.User{
-				Uid:      "1337",
+				Uid:      fmt.Sprintf("%d", os.Getuid()),
```

Comment:
That looks to me that something else needed to be mocked instead. Or we have some kind of side effect.

Reply:
I did this to avoid keeping two values synced, whatever this value is - must match the value provided in api_apps_test.go:183 `xdgRuntimeDir := fmt.Sprintf("%s/%d", dirs.XdgRuntimeDirBase, os.Getuid())`. The problem is that currently, due to the user-session agent running as a part of the unit test, this must run under the user that executes the test.

Reply:
Maybe we should mock the user-session agent, or at least make `os.Getuid` (there are other tests that do this I think).

Reply:
I'll just investigate this more closely later - I might have missed something here